### ISSUE TITLE
Keyed Messages - Expects partition?

### ISSUE THREAD

**Issue by peek4y** [oleksiyk/kafka#67]
AFAIK, while sending out keyed messages, it should automatically be distributed to a partition based on the key, and I guess we don't need to specify the partition explicitly? 

So, while sending out keyed messages, via ```no-kafka```  it's expecting a partition, I get an error ```Missing or invalid partition```.

Is this expected?

**Comment by oleksiyk** [comment]
There is no default partitioner in no-kafka 2.x so this is indeed expected behaviour. 

This is addressed in upcoming 3.0 branch by using `Kafka.DefaultPartitioner` which matches Java client implementation.

For 2.x just use a simple round robin partitioner:

```javascript
partitioner: function (_topicName, partitions, message) {
  var crc = crc32(message.key);
  return crc % partitions.length;
}
```

**Comment by peek4y** [comment]
Cool thanks!
